terraform/deployments/monitoring-test/base/main.tf

```
@@ -0,0 +1,71 @@
+terraform {
+  backend "s3" {
+    bucket  = "govuk-terraform-test"
+    key     = "projects/monitoring.tfstate"
+    region  = "eu-west-1"
+    encrypt = true
+  }
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.13"
+    }
+  }
+}
+
+provider "aws" {
+  region = "eu-west-1"
+
+  assume_role {
+    role_arn = var.assume_role_arn
+  }
+}
+
+data "terraform_remote_state" "infra_networking" {
+  backend = "s3"
+  config = {
+    bucket   = var.govuk_aws_state_bucket
+    key      = "govuk/infra-networking.tfstate"
+    region   = "eu-west-1"
+    role_arn = var.assume_role_arn
+  }
+}
+
+data "terraform_remote_state" "infra_security_groups" {
+  backend = "s3"
+  config = {
+    bucket   = var.govuk_aws_state_bucket
+    key      = "govuk/infra-security-groups.tfstate"
+    region   = "eu-west-1"
+    role_arn = var.assume_role_arn
+  }
+}
+
+data "terraform_remote_state" "govuk" {
+  backend = "s3"
+  config = {
+    bucket   = "govuk-terraform-test"
```

```suggestion
    bucket   = var.govuk_aws_state_bucket
```

If `var.govuk_aws_state_bucket` is permitted in both `data.config` and `terraform.backend` blocks, this would make the module reusable by other environments.

We could move the module and pass in the environment specific variables:

```diff
+ terraform/deployments/monitoring-test/base/main.tf
- terraform/deployments/monitoring/main.tf
```

What do you think?

---

yeah, I was using govuk-test as the example but I can do that because I think this will be the new style. I don't know if I want to keep the base folder though.

---

Thanks. Yes I think the `base` nesting is unnecessary.

---

since the backend is not parameterizable, we will have to stick to a fixed value.